flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/planner/plan/stream/sql/TableScanTest.xml

```diff
@@ -556,10 +577,10 @@ LogicalProject(a=[$1], b=[$2], c=[$3])
     <Resource name="planAfter">
       <![CDATA[
 Calc(select=[a, b, c], where=[>(a, 1)], changelogMode=[I,UB,UA,D])
-+- WatermarkAssigner(rowtime=[ts], watermark=[-(ts, 1000:INTERVAL SECOND)], changelogMode=[I,UB,UA,D])
-   +- Calc(select=[id, a, +(a, 1) AS b, c, TO_TIMESTAMP(c) AS ts], changelogMode=[I,UB,UA,D])
-      +- ChangelogNormalize(key=[id], changelogMode=[I,UB,UA,D])
-         +- Exchange(distribution=[hash[id]], changelogMode=[UA,D])
++- ChangelogNormalize(key=[id], changelogMode=[I,UB,UA,D])
+   +- Exchange(distribution=[hash[id]], changelogMode=[UA,D])
+      +- WatermarkAssigner(rowtime=[ts], watermark=[-(ts, 1000:INTERVAL SECOND)], changelogMode=[UA,D])
+         +- Calc(select=[id, a, +(a, 1) AS b, c, TO_TIMESTAMP(c) AS ts], changelogMode=[UA,D])
```

Comment:
It's a bad case here, that more data needs to shuffle now.

Reply:
I think this can be an improvement. Because `ChangelogNormalize` along with the upsert source is introduced in this version, there is no regressions. We can pull out the `+(a, 1)` to be after the `ChangelogNormalize` in the future. I have created https://issues.apache.org/jira/browse/FLINK-20289 to track this improvement.